### ISSUE TITLE
Reenable ommitted tests for Windows

### ISSUE THREAD

**Issue by martukas** [ess-dmsc/h5cpp#249]
* Symlinks require admin privileges in Windows. As such, tests that use them cannot run on Jenkins. The affected test suites are FileCreation and ObjectId. We might want to re-enable these at some point in the future.
* For some reason, h5py string compatibility tests fail on Windows, with the following exceptions:
```
[----------] 2 tests from H5pyStringCompatTest
[ RUN      ] H5pyStringCompatTest.test_read_scalar_string
unknown file: error: C++ exception with description "Invalid object handler (-1)" thrown in SetUp().
[  FAILED  ] H5pyStringCompatTest.test_read_scalar_string (4 ms)
[ RUN      ] H5pyStringCompatTest.test_read_vector_string
unknown file: error: C++ exception with description "Invalid object handler (-1)" thrown in SetUp().
[  FAILED  ] H5pyStringCompatTest.test_read_vector_string (0 ms)
[----------] 2 tests from H5pyStringCompatTest (4 ms total)
```

**Comment by zjttoefs** [comment]
To be looked at during rewrite of tests.

**Comment by jkotan** [comment]
It looks like after the test upgrades the only windows omitted tests are:
- a link access test, i.e. addressed in #544
- three tests using testing access to files connected via symbolic links (2 in `core/object_id_test` and 1 in file/file_creation_test.cpp)

On Windows 10 it is possible to give rights to a user to create symbolic links. However, it is not recommended https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/create-symbolic-links
because symbolic links on windows system can be abused to cause elevation of privilege or arbitrary write/delete.
Thus, it is better to not encourage to use them on windows and it is better to keep the windows symlinks tests disabled.

Therefore I think after merging #544 we can close the tickets.
